### ISSUE TITLE
Insufficient contrast on borders

### ISSUE THREAD

**Issue by xi** [twbs/bootstrap#38480]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

[WCAG 2.1](https://www.w3.org/TR/WCAG21/#non-text-contrast) requires a contrast of 3:1 for non-text content. Many elements, e.g. form inputs or checkboxes, mostly consist of borders, at least in some of their states. It is therefore important that the border color has sufficient contrast to the background.

[`#888888`](https://contrast-ratio.com/#%23999-on-white) would have sufficient contrast to white. Bootstrap 5.2 uses [`#ced4da`](https://contrast-ratio.com/#%23ced4da-on-white), which only has a contrast of 1.49 to white. Bootstrap 5.3 takes this even further by using [`#dee2e6`](https://contrast-ratio.com/#%23dee2e6-on-white) which only has a contrast of 1.3 to white.

I do not have any vision impairments that I know of, but to me these borders are practically invisible (depending on the lighting conditions). I strongly recommend to revert at least to the level of Bootstrap 5.2.

### Reduced test cases

![#888888](https://user-images.githubusercontent.com/202576/232973631-a415ac4e-611b-4064-b23e-b36daa5d525f.png)
![Bootstrap 5.2.3](https://user-images.githubusercontent.com/202576/232973636-9a4e3ef4-bd48-46df-8479-d3e1e5dfd8a3.png)
![Bootstrap 5.3.0-alpha3](https://user-images.githubusercontent.com/202576/232973635-274cc544-2c0f-48a1-90db-7efc05dd1a84.png)

### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

5.3.0-alpha3

**Comment by raghav1030** [comment]
is the issue still open ? if so I would very much like to contribute to it?

**Comment by ChellyAhmed** [comment]
Hello,
Thank you @xi for opening this issue. I also think it is relevant to consider the contrast ratio for the border colors. I used Adobe Color tool (https://color.adobe.com/create/color-contrast-analyzer) for double-checking the contrast. As you mentioned a 1.3:1 to white is not recommended. The lightest color to achieve a ratio of 3.01 : 1 that was suggested by the same tool would be `#929598`. But I am not sure about which colors we are allowed to use in Bootstrap in case we change the color. Even if we do so, will this mean that we need to update the colors of other elements?

**Comment by xi** [comment]
Reacting to https://github.com/twbs/bootstrap/pull/38489#issuecomment-1518506999:

> Let's keep the PRs at bay for now and discuss any potential changes in the issue. I could see us darkening things a little, but this is too much

I think this regression happened in [fc3f4b67](https://github.com/twbs/bootstrap/commit/fc3f4b67d65c575daa661ecf31cf59b4ff3cced5#diff-108a57a50d5ebe203f447ef7bf1e5e53826bdb82baab2baa2b41b4158cdd4d09R873):

- `$input-border-color` was changed from `$gray-400` to `var(--border-color)` which defaults to `$gray-300`
- the color in `$form-check-input-border` was changed from `rgba($black, .25)` also to `var(--#{$prefix}border-color)`

I see two options to fix this:

- Add a new CSS variable `--input-border-color` and set it to `$gray-400`
- Change `--border-color` to `$gray-400`

@mdo which option do you prefer?

Btw, I also checked the contrast for dark mode ([`#495057` on `#212529`](https://contrast-ratio.com/#%23495057-on-%23212529)) and it fares much better: 1.88 is still too low, but much better than the contrasts in light mode.

**Comment by ChellyAhmed** [comment]
Hello @xi 
Thank you for mentioning that the colors changed from Bootstrap 5.2 to 5.3 resulting in a non-recommended contrast ratio.
But I think it is relevant to check why the colors were changed in the first place so we know if changing color is allowed, and the side effects of such a decision. Do you have any input about that?

**Comment by RobineSavert** [comment]
Please a fix for this, as the checkboxes are now almost not visible in light mode

**Comment by jprante** [comment]
Please,  a quick fix is needed. Light mode is almost unusable!

**Comment by jasontconnell** [comment]
> I see two options to fix this:
> 
> * Add a new CSS variable `--input-border-color` and set it to `$gray-400`
> * Change `--border-color` to `$gray-400`

--input-border-color probably has to be it. I set the border color globally and it was too drastic of a change on all of the components. Being able to have an --input-border-color variable would be ideal. It has been a while since this comment, but the issue is still open, so hopefully it's still a consideration. I see the issue moved to 5.3.4 also.

**Comment by mdo** [comment]
Maybe we adjust in v5.4.0.

**Comment by nwalters512** [comment]
I reported this in #41414; unfortunately this issue didn't show up in my searches.

My team is getting reports that this change is having actual negative impact on our end users. I'd love to see this get fixed.
